File: two_sum/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) ->list[int]:
        table = {}
        for i in enumerate(nums):
            if (table.__contains__(target - i[1])): return [table[target - i[1]], i[0]]
            table[i[1]] = i[0]
        return [-1, -1]
        
        
        
nums = [0, 0]
target = 0

sol = Solution()
print(sol.twoSum(nums, target))

File: two_sum/two_sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector <int> twoSum(vector <int> & nums, int target) {
        unordered_map <int, int> table;

        for (auto itr = nums.begin(); itr != nums.end(); itr++){
            auto got = table.find(target - *itr);
            int index = itr - nums.begin();

            if (got == table.end()) table[*itr] = index;
            else return {got->second, index};
        }
        return {0, 0};
    }
};

int main(void){

    vector <int> nums = {2, 5, 5, 11};
    int target = 10;

    Solution sol;

    auto ans = sol.twoSum(nums, target);

    cout << "[" << ans[0] << "," << ans[1] << "]" << "\n";
}